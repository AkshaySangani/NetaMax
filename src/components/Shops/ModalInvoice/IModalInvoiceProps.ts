import { IModalInvoice } from "dataflows/Shops/IModalInvoice";

export interface IModalInvoiceProps {
  /**
   * Payload of getModalInvoice()
   * @type {IModalInvoice}
   * */
  modalInvoice: IModalInvoice;

  credit: {
    InvoiceType: number;
    creditToPay: number;
    dayOfWeek: number;
    dropToPay: number;
    totalToPayToday: number;
  };

  invoice: {
    createdOnUtc: string;
    customerDebit: number;
    deliveryStatus: string;
    id: string;
    invoiceDate: string;
    orderDisccount: number;
    orders: string;
    pdfUrl: string;
    qtyCustomers: number;
    qtyOrders: number;
    status: string;
    storeCommisssion: number;
    storeDebit: number;
    storeId: string;
    storeName: string;
    storeOwner: string;
    storePhone: string;
    subTotal: number;
  };

  metadata: {
    currentPage: number;
    nextRefresh: number;
    perPage: number;
    totalPages: number;
    updatedAt: string;
  };

  orderItems: Array<{
    PriceInclTax: number;
    Quantity: number;
    UnitPriceInclTax: number;
    id: number;
    idProduct: number;
    name: string;
  }>;

  storeProfit: {
    historical: number;
    month: number;
    profitmonthdatefrom: string;
    profitweekdatefrom: string;
    today: number;
    week: number;
    profitweekdateto: string;
  };
  /**
   * indicate if Modal is open
   * @type {boolean}
   * */
  isOpenModal: boolean;
  /**
   * function to close the ModalInvoice
   * @type {void}
   * */
  onClose: () => void;
  /**
   * function to open the ModalInvoice
   * @type {void}
   * */
  onOpen: () => void;
}
