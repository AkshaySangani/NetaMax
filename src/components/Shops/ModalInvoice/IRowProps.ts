export interface IRowProps {
  id: number;
  Quantity: number;
  name: string;
  UnitPriceInclTax: number;
  PriceInclTax: number;
}
