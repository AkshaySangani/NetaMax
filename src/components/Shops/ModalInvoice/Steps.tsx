import React, { ReactElement } from "react";

import { Box, Flex, Text } from "@chakra-ui/react";
/**
 * Steps component of ModalInvoice
 * @param {IModalInvoiceProps} props Shops Menu props
 * @returns {ReactElement} The menu container
 */
export const Steps = (): ReactElement => {
  return (
    <Box w="100%" bg="#fff">
      <Box bg="#F0F5FF" rounded="lg" p={["1.4rem", "1.8rem"]} mr="6" ml="6">
        <Flex
          direction={["column", "column", "row"]}
          fontSize={{ base: "13px", md: "13px", lg: "14px" }}
          gap="1.5rem"
        >
          <Flex>
            <Text
              fontSize={{ base: "17px", md: "18px", lg: "20px" }}
              fontWeight="700"
              lineHeight={["16.5px", "26.5px"]}
              mr={["3px", "8px"]}
              color="#68717D"
            >
              1.
            </Text>
            <Text color="#767676">
              <span>
                <b>Revisa tu orden</b> al momento de la entrega y solo pagar los productos que
                recibas
              </span>
            </Text>
          </Flex>
          <Flex>
            <Text
              fontSize={{ base: "17px", md: "18px", lg: "20px" }}
              fontWeight="700"
              lineHeight={["16.5px", "26.5px"]}
              mr={["3px", "8px"]}
              color="#68717D"
            >
              2.
            </Text>
            <Text color="#767676">
              <span>
                <b>Tienes 24 horas</b> después de la entrega para hacer cualquier reclamación
              </span>
            </Text>
          </Flex>
          <Flex>
            <Text
              fontSize={{ base: "17px", md: "18px", lg: "20px" }}
              fontWeight="700"
              lineHeight={["16.5px", "26.5px"]}
              mr={["3px", "8px"]}
              color="#68717D"
            >
              3.
            </Text>
            <Text color="#767676">
              <span>
                <b>Si deseas hacer algun cambio</b> en tu entrega por favor comunícate al 5545439866
              </span>
            </Text>
          </Flex>
        </Flex>
      </Box>
    </Box>
  );
};
