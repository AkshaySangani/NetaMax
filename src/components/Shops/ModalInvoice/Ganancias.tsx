import React, { ReactElement } from "react";

import { Box, Flex, Stack, Text } from "@chakra-ui/react";
import newformatDate from "../../../utils/dateUtils";
import { IModalInvoiceProps } from "./IModalInvoiceProps";
import { toFixed } from "../../../utils/numberUtils";
/**
 * The Ganancias component of ModalInvoice
 * @param {IModalInvoiceProps} props Shops Menu props
 * @returns {ReactElement} The menu container
 */
export const Ganancias = (props: IModalInvoiceProps): ReactElement => {
  const { invoice, storeProfit } = props;

  const { profitweekdatefrom, profitweekdateto, month, week, historical } = storeProfit;

  const today = new Date(invoice.invoiceDate);
  const newDateFormatWeekFrom = `${newformatDate(profitweekdatefrom, true)}`;
  const newDateFormatWeekTo = `${newformatDate(profitweekdateto, true)}`;
  const newDateFormatDay = `${newformatDate(today, true)}`;

  return (
    <Stack
      direction={["column-reverse", "column-reverse", "column-reverse", "row"]}
      pt="7"
      pb="12"
      gap="6"
      pl="6"
      pr="6"
      bg="#fff"
    >
      <Box
        w={["100%", "100%", "100%", "45%"]}
        h="165px"
        backgroundImage="linear-gradient(180deg, #dde8ff 2.50%, #f6f9ff 2.50%, #f6f9ff 50%, #dde8ff 50%, #dde8ff 52.50%, #f6f9ff 52.50%, #f6f9ff 100%)"
        backgroundSize="68.00px 68.00px"
        borderRadius="12"
      >
        <Flex justify="space-around" h="100%">
          <Flex direction="column" justify="flex-end">
            <Text color="#82B2FF" fontWeight="700" textAlign="center">
              {toFixed(storeProfit.today)}
            </Text>
            <Box w="45px" h="25px" bg="#BBD5FF" borderTopRadius="6"></Box>
          </Flex>
          <Flex direction="column" justify="flex-end">
            <Text color="#FFA947" fontWeight="700" textAlign="center">
              {toFixed(week)}
            </Text>
            <Box w="45px" h="55px" bg="#FFD5A6" borderTopRadius="6"></Box>
          </Flex>
          <Flex direction="column" justify="flex-end" alignItems="center">
            <Text color="#92C048" fontWeight="700" textAlign="center">
              {toFixed(month)}
            </Text>
            <Box w="45px" h="90px" bg="#CCEB98" borderTopRadius="6"></Box>
          </Flex>
          <Flex direction="column" justify="flex-end" alignItems="center">
            <Text color="#FF6C96" fontWeight="700" textAlign="center">
              {toFixed(historical)}
            </Text>
            <Box w="45px" h="130px" bg="#FFB5CA" borderTopRadius="6"></Box>
          </Flex>
        </Flex>
        <Flex fontSize={["13px", "15px"]}>
          <Text w="25%" align="center">
            Hoy
          </Text>
          <Text w="25%" align="center">
            Semana
          </Text>
          <Text w="25%" align="center">
            {newDateFormatDay.split(" ")[2]}
          </Text>
          <Text w="25%" align="center">
            Históricas
          </Text>
        </Flex>
      </Box>
      <Box fontSize="13.5px">
        <Text fontSize={["16px", "15px"]} fontWeight="700" pb="4">
          Ganancias de mi tienda
        </Text>
        <Flex color="#767676" direction="column" gap="3">
          <Flex gap="5px">
            <Box w={["12px", "14px"]} h={["12px", "14px"]} bg="#BBD5FF" alignSelf="center"></Box>
            <Text>
              <span>
                <b>Hoy</b> ({newDateFormatDay}) Ganaste ${toFixed(storeProfit.today)}
              </span>
            </Text>
          </Flex>

          <Flex gap="5px">
            <Box w={["12px", "14px"]} h={["12px", "14px"]} bg="#FFD5A6" alignSelf="center"></Box>
            <Text>
              <span>
                <b>Semana</b> (
                {newDateFormatWeekFrom.split(" ")[0] +
                  " " +
                  newDateFormatWeekFrom.split(" ")[1] +
                  " " +
                  newDateFormatWeekFrom.split(" ")[2] +
                  " / " +
                  newDateFormatWeekTo}
                ) Ganaste ${toFixed(week)}
              </span>
            </Text>
          </Flex>

          <Flex gap="5px">
            <Box w={["12px", "14px"]} h={["12px", "14px"]} bg="#CCEB98" alignSelf="center"></Box>
            <Text>
              <span>
                <b>Mes de {newDateFormatDay.split(" ")[2]}</b> - Ganaste ${toFixed(month)}
              </span>
            </Text>
          </Flex>

          <Flex gap="5px">
            <Box w={["12px", "14px"]} h={["12px", "14px"]} bg="#FFB5CA" alignSelf="center"></Box>
            <Text>
              <span>
                <b>Históricas</b> (inicio con Neta a hoy) Ganaste ${toFixed(historical)}
              </span>
            </Text>
          </Flex>
        </Flex>
      </Box>
    </Stack>
  );
};
