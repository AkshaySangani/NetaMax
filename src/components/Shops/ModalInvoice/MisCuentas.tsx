import React, { ReactElement } from "react";

import { screenSizes } from "styled/screen";

import { useMediaQuery, Box, Center, Divider, Flex, Text } from "@chakra-ui/react";

import { IModalInvoiceProps } from "./IModalInvoiceProps";
/**
 * MisCuentas component of ModalInvoice
 * @param {IModalInvoiceProps} props Shops Menu props
 * @returns {ReactElement} The menu container
 */
export const MisCuentas = (props: IModalInvoiceProps): ReactElement => {
  const [isDesktop] = useMediaQuery(`(min-width: ${screenSizes.mdMin}px)`);
  const { invoice } = props;
  const { invoiceDate, id, storeName, storeOwner } = invoice;

  return (
    <Box pt={{ base: "5%", md: "4%", lg: "3%" }} bg="#F5F5F5" borderTopRadius="inherit">
      <Text
        fontSize={{ base: "21px", md: "25px", lg: "25px" }}
        fontWeight="700"
        textAlign="left"
        pl="5%"
        pb={{ base: "2%", md: "3%", lg: "1%" }}
        color="#051549"
      >
        Mis cuentas
      </Text>
      <Divider orientation="horizontal" />
      <Flex
        color="#4A4A4A"
        w="100%"
        align="center"
        justifyContent={["flex-start", "flex-start", "flex-start", "center"]}
        py={{ base: "5%", md: "4%", lg: "3%" }}
      >
        {!isDesktop ? (
          <Flex direction="column" justifyContent="left" pl="6%" gap="3">
            <Text fontSize={{ base: "14px", md: "17px", lg: "16px" }} fontWeight={400}>
              Número de factura&nbsp;&nbsp;&nbsp;
              <span style={{ fontWeight: "600" }}>&#124;&nbsp;&nbsp; {id}</span>
            </Text>
            <Text fontSize={{ base: "14px", md: "17px", lg: "16px" }} fontWeight={400}>
              Fecha de factura&nbsp;&nbsp;&nbsp;
              <span style={{ fontWeight: "600" }}>&#124;&nbsp;&nbsp; {invoiceDate}</span>
            </Text>
            <Text fontSize={{ base: "14px", md: "17px", lg: "16px" }} fontWeight={400}>
              Tienda&nbsp;&nbsp;&nbsp;
              <span style={{ fontWeight: "600" }}>&#124;&nbsp;&nbsp; {storeName}</span>
            </Text>
            <Text fontSize={{ base: "14px", md: "17px", lg: "16px" }} fontWeight={400}>
              Propietario&nbsp;&nbsp;&nbsp;
              <span style={{ fontWeight: "600" }}>&#124;&nbsp;&nbsp; {storeOwner}</span>
            </Text>
          </Flex>
        ) : (
          <Flex direction="row" justifyContent="center" gap="7rem" fontSize="15px" fontWeight="400">
            <Flex gap="2rem">
              <Flex direction="column" justifyContent="center" gap="1rem">
                <Text>Número de factura</Text>
                <Text>Fecha de factura</Text>
              </Flex>
              <Center height="90%" alignItems="baseline">
                <Divider orientation="vertical" borderWidth="2px" color="#E1E1E1" />
              </Center>
              <Flex direction="column" justifyContent="center" fontWeight="600" gap="1rem">
                <Text>{id}</Text>
                <Text>{invoiceDate}</Text>
              </Flex>
            </Flex>

            <Flex gap="2rem">
              <Flex direction="column" justifyContent="center" gap="1rem">
                <Text>Tienda</Text>
                <Text>Propietario</Text>
              </Flex>
              <Center height="90%" alignItems="baseline">
                <Divider orientation="vertical" borderWidth="2px" color="#E1E1E1" />
              </Center>
              <Flex direction="column" justifyContent="center" fontWeight="600" gap="1rem">
                <Text>{storeName}</Text>
                <Text>{storeOwner}</Text>
              </Flex>
            </Flex>
          </Flex>
        )}
      </Flex>
    </Box>
  );
};
