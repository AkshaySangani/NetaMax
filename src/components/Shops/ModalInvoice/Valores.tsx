import React, { ReactElement } from "react";

import { screenSizes } from "styled/screen";

import { useMediaQuery, Box, Divider, Flex, Text, Button } from "@chakra-ui/react";
import IconWhatsapp from "styled/icons/Shops/Order/Whatsapp";
import IconCurrency from "styled/icons/Shops/Summary/Currency";
import IconCash from "styled/icons/Shops/Summary/Cash";
import TruckSolid from "styled/icons/Shops/Summary/TruckSolid";
import NetaIcon from "styled/icons/Shops/Summary/NetaIcon";
import newformatDate from "../../../utils/dateUtils";
import { IModalInvoiceProps } from "./IModalInvoiceProps";
/**
 * The Valores component of ModalInvoice
 * @param {IModalInvoiceProps} props Shops Menu props
 * @returns {ReactElement} The menu container
 */
export const Valores = (props: IModalInvoiceProps): ReactElement => {
  const { metadata, credit, invoice } = props;
  const [isDesktop] = useMediaQuery(`(min-width: ${screenSizes.mdMin}px)`);

  const { updatedAt } = metadata;
  const { dropToPay, creditToPay, totalToPayToday } = credit;
  const { storeDebit } = invoice;

  const newDateFormatDay = `${newformatDate(updatedAt, true)}`;

  return (
    <Box bg="#fff" pt="10" pb="6" pl="4" pr="4">
      <Flex
        direction={["column", "column", "row"]}
        justifyContent="center"
        gap={["1.5rem", "2rem"]}
        color="#939393"
      >
        <Flex d="column" textAlign="center" justifyContent="center">
          <Flex justifyContent="center" pb="2" gap="1rem">
            <TruckSolid />
            <Text lineHeight="30px" fontSize={["14px", "15px", "17px"]}>
              ${dropToPay}
            </Text>
          </Flex>
          <Flex justifyContent="center" fontSize={["13px", "14px", "15px"]}>
            Valor total de la entrega
          </Flex>
        </Flex>
        {isDesktop && <Divider h="3rem" orientation="vertical" borderWidth="1px" color="#BABABA" />}

        <Flex justifyContent="center" gap={["2rem"]}>
          <Flex w="50%" d="column" textAlign="center">
            <Flex justifyContent="center" pb="2" gap="1rem">
              <IconCash />
              <Text lineHeight="30px" fontSize={["14px", "15px", "17px"]}>
                ${creditToPay}
              </Text>
            </Flex>
            <Flex justifyContent="center" fontSize={["13px", "14px", "15px"]}>
              Saldo adeudado en Crédito al {newDateFormatDay}
            </Flex>
          </Flex>
          <Divider h={["4rem", "3rem"]} orientation="vertical" borderWidth="1px" color="#BABABA" />
          <Flex d="column" textAlign="center">
            <Flex justifyContent="center" pb="2" gap="1rem">
              <NetaIcon />
              <Text fontSize={["14px", "15px", "17px"]} lineHeight="30px">
                ${storeDebit}
              </Text>
            </Flex>
            <Flex justifyContent="center" fontSize={["13px", "14px", "15px"]}>
              Débito de tu Billetera Neta
            </Flex>
          </Flex>
        </Flex>
      </Flex>
      <Divider
        ml="auto"
        mr="auto"
        justify="center"
        w="80%"
        mt={["6", "8"]}
        orientation="horizontal"
        borderWidth="1px"
        color="#BABABA"
      />
      <Flex pb="7" pt="4" alignItems="center" color="#333333" fontWeight="700">
        <Flex width={["50%", "40%"]} justifyContent="flex-end" pr="6">
          <IconCurrency />
          <Text lineHeight="2" pl="3" fontSize={["17px", "18px"]}>
            {totalToPayToday}
          </Text>
        </Flex>
        <Divider
          borderColor="#333333"
          h={["2rem", "1rem"]}
          orientation="vertical"
          borderWidth="1px"
        />
        <Text
          justifyContent="flex-start"
          fontSize={["13px", "16px"]}
          textAlign="start"
          pl={["1rem", "2rem"]}
        >
          Total a pagar al conductor en Efectivo
        </Text>
      </Flex>
      <Flex justify="center">
        <Button
          leftIcon={<IconWhatsapp />}
          background="#03CA69"
          color="#FDFCFE"
          _hover={{ bg: "#028545" }}
          borderRadius="99px"
        >
          <Text fontSize={["12px", "14px"]}>Si tienes dudas contáctanos</Text>
        </Button>
      </Flex>
    </Box>
  );
};
