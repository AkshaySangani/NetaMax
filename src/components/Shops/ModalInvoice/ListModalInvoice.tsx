import React, { ReactElement } from "react";

import { Box, Text, Table, Tbody, Tr, Td } from "@chakra-ui/react";

import { IModalInvoiceProps } from "./IModalInvoiceProps";
import { columns } from "../../../constants/shopsConstants";
import { IRowProps } from "./IRowProps";
/**
 * ListModal component of ModalInvoice
 * @param {IModalInvoiceProps} props Shops Menu props
 * @returns {ReactElement} The menu container
 */
export const List = (props: IModalInvoiceProps): ReactElement => {
  const { orderItems } = props;

  return (
    <Box width={["100%", "90%"]} ml="auto" mr="auto">
      <Text textAlign={["center", "start"]} fontSize={["12px", "15px"]} bg="#fff" p="4">
        Este es el acumulado de productos solicitados por tus clientes.
      </Text>
      <div>
        <Table variant="striped" colorScheme="gray">
          <thead>
            <Tr>
              {columns.map((head) => (
                <th
                  key={head.id}
                  style={{ fontSize: "13px", textAlign: "center", color: "#767676" }}
                >
                  {head.header}
                </th>
              ))}
            </Tr>
          </thead>
          <Tbody>
            {orderItems.map((row: IRowProps) => (
              <Tr key={row.id}>
                <Td color="#767676" fontSize={["13px", "15px"]} textAlign="center">
                  {row.Quantity}
                </Td>
                <Td pl="0" pr="0" color="#767676" fontSize={["13px", "15px"]} textAlign="start">
                  {row.name}
                </Td>
                <Td color="#767676" fontSize={["13px", "15px"]} textAlign="center">
                  ${row.UnitPriceInclTax}
                </Td>
                <Td color="#767676" fontSize={["13px", "15px"]} textAlign="center">
                  ${row.PriceInclTax}
                </Td>
                <Td>
                  <input type="checkbox"></input>
                </Td>
              </Tr>
            ))}
          </Tbody>
        </Table>
      </div>
    </Box>
  );
};
