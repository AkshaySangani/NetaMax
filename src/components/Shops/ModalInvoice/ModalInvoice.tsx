import React, { ReactElement } from "react";
import { screenSizes } from "styled/screen";
import { MisCuentas } from "./MisCuentas";
import { Steps } from "./Steps";
import { Valores } from "./Valores";
import { List } from "./ListModalInvoice";
import { Ganancias } from "./Ganancias";
import {
  SA_VOLVER_BUTTON_CLICKED,
  SA_DESCARGAR_BUTTON_CLICKED,
  TrackerApp,
} from "constants/amplitudeConstants";
import { trackEvent } from "utils/tracker";

import {
  useMediaQuery,
  Center,
  Modal,
  ModalBody,
  ModalCloseButton,
  ModalContent,
  ModalOverlay,
  Button,
  Stack,
  Link,
} from "@chakra-ui/react";
import { IModalInvoiceProps } from "./IModalInvoiceProps";
/**
 * Modal in Summary component
 * @param {IModalInvoiceProps} props Shops Menu props
 * @returns {ReactElement} The menu container
 */
export const ModalInvoice = ({
  isOpenModal,
  onClose,
  onOpen,
  invoice,
  credit,
  metadata,
  orderItems,
  storeProfit,
  modalInvoice,
}: IModalInvoiceProps): ReactElement => {
  const [isDesktop] = useMediaQuery(`(min-width: ${screenSizes.mdMin}px)`);

  return (
    modalInvoice && (
      <Center>
        <Modal
          isOpen={isOpenModal}
          onClose={onClose}
          size={isDesktop ? "4xl" : "md"}
          scrollBehavior="inside"
        >
          <ModalOverlay />
          <ModalContent maxH="100%" ml="2%" mr="2%" pt="0" rounded="lg" gap="1.5rem">
            <ModalCloseButton fontSize="13px" background="transparent" />
            <ModalBody w="100%" gap="1.5rem" p="0" rounded="inherit" borderColor="#C9C9C9">
              <MisCuentas
                invoice={invoice}
                credit={credit}
                metadata={metadata}
                orderItems={orderItems}
                storeProfit={storeProfit}
                modalInvoice={modalInvoice}
                onOpen={onOpen}
                onClose={onClose}
                isOpenModal={isOpenModal}
              />
              <Ganancias
                invoice={invoice}
                credit={credit}
                metadata={metadata}
                orderItems={orderItems}
                storeProfit={storeProfit}
                modalInvoice={modalInvoice}
                onOpen={onOpen}
                onClose={onClose}
                isOpenModal={isOpenModal}
              />
              <Steps />
              <Valores
                invoice={invoice}
                credit={credit}
                metadata={metadata}
                orderItems={orderItems}
                storeProfit={storeProfit}
                modalInvoice={modalInvoice}
                onOpen={onOpen}
                onClose={onClose}
                isOpenModal={isOpenModal}
              />
              <List
                invoice={invoice}
                credit={credit}
                metadata={metadata}
                orderItems={orderItems}
                storeProfit={storeProfit}
                modalInvoice={modalInvoice}
                onOpen={onOpen}
                onClose={onClose}
                isOpenModal={isOpenModal}
              />
              <Stack
                w={["100%", "90%"]}
                justify={["center", "flex-end"]}
                direction="row"
                spacing={8}
                align="center"
                pt="5"
                pb="40"
                mr="auto"
                ml="auto"
              >
                <Button
                  colorScheme="#002F7A"
                  variant="outline"
                  borderWidth="2px"
                  borderColor="#D5D5D5"
                  borderRadius="99px"
                  pl="10"
                  pr="10"
                  onClick={() => {
                    onClose();
                    trackEvent(SA_VOLVER_BUTTON_CLICKED, [
                      TrackerApp.Amplitude,
                      TrackerApp.Segment,
                    ]);
                  }}
                >
                  Volver
                </Button>
                <Link
                  href={invoice.pdfUrl === "" ? undefined : invoice.pdfUrl}
                  target="_blank"
                  data-testid="Factura link"
                  _hover={{
                    textDecoration: "none",
                    cursor: invoice.pdfUrl === "" ? "default" : "pointer",
                  }}
                >
                  <Button
                    colorScheme="#002F7A"
                    variant="outline"
                    borderWidth="2px"
                    borderColor="#D5D5D5"
                    borderRadius="99px"
                    pl="10"
                    pr="10"
                    onClick={() =>
                      trackEvent(SA_DESCARGAR_BUTTON_CLICKED, [
                        TrackerApp.Amplitude,
                        TrackerApp.Segment,
                      ])
                    }
                  >
                    Descargar
                  </Button>
                </Link>
              </Stack>
            </ModalBody>
          </ModalContent>
        </Modal>
      </Center>
    )
  );
};
