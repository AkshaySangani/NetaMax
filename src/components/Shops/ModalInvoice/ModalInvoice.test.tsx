import { render, cleanup } from "@testing-library/react";
import { ModalInvoice } from "./ModalInvoice";
import { IModalInvoiceProps } from "./IModalInvoiceProps";

jest.mock("next/router", () => require("next-router-mock"));
let mock: IModalInvoiceProps;

describe("ModalInvoice Component", () => {
  const onClose = jest.fn();
  const onOpen = jest.fn();
  const modalInvoice = {
    invoice: {
      id: "20014",
      storeId: "4221",
      storeName: "Venta de catalogo Gina ",
      storePhone: "5584497063",
      storeOwner: "Georgina angel mora",
      pdfUrl:
        "https://ops-facturas.s3.us-east-2.amazonaws.com/Facturas/2022/04/04/stores/4221/2022-04-04-4221-invoice",
      status: "CLOSED",
      orders: "234602,234605,234618,236841,236848,236850,236851",
      invoiceDate: "2022-04-04",
      subTotal: 1746.8126,
      orderDisccount: 99,
      customerDebit: 0,
      storeCommisssion: 70,
      storeDebit: 0,
      createdOnUtc: "2022-04-05T10:19:02.000Z",
      deliveryStatus: "PENDING",
      qtyOrders: 7,
      qtyCustomers: 7,
    },
    credit: {
      dropToPay: 1746.8126,
      creditToPay: 0,
      totalToPayToday: 1746.8126,
      dayOfWeek: 1,
      InvoiceType: 0,
    },
    storeProfit: {
      today: 0,
      week: 0,
      month: 170,
      historical: 1350,
      profitweekdatefrom: "2022-04-04T00:00:00.000Z",
      profitmonthdatefrom: "2022-04-01T00:00:00.000Z",
      profitweekdateto: "2022-04-04 23:59:00",
    },
    orderItems: [
      {
        id: 615724,
        idProduct: 65,
        name: "Antitranspirante Rexona Motion Sense powder dry en roll on para dama 50 ml",
        Quantity: 1,
        UnitPriceInclTax: 20.9,
        PriceInclTax: 20.9,
      },
    ],
    metadata: {
      currentPage: 1,
      perPage: 1,
      totalPages: 1,
      updatedAt: "2022-05-27T18:28:48+00:00",
      nextRefresh: 3600,
    },
  };

  beforeEach(() => {
    mock = {
      onClose,
      onOpen,
      modalInvoice,
      invoice: {
        id: "20014",
        storeId: "4221",
        storeName: "Venta de catalogo Gina ",
        storePhone: "5584497063",
        storeOwner: "Georgina angel mora",
        pdfUrl:
          "https://ops-facturas.s3.us-east-2.amazonaws.com/Facturas/2022/04/04/stores/4221/2022-04-04-4221-invoice",
        status: "CLOSED",
        orders: "234602,234605,234618,236841,236848,236850,236851",
        invoiceDate: "2022-04-04",
        subTotal: 1746.8126,
        orderDisccount: 99,
        customerDebit: 0,
        storeCommisssion: 70,
        storeDebit: 0,
        createdOnUtc: "2022-04-05T10:19:02.000Z",
        deliveryStatus: "PENDING",
        qtyOrders: 7,
        qtyCustomers: 7,
      },
      credit: {
        dropToPay: 1746.8126,
        creditToPay: 0,
        totalToPayToday: 1746.8126,
        dayOfWeek: 1,
        InvoiceType: 0,
      },
      storeProfit: {
        today: 0,
        week: 0,
        month: 170,
        historical: 1350,
        profitweekdatefrom: "2022-04-04T00:00:00.000Z",
        profitmonthdatefrom: "2022-04-01T00:00:00.000Z",
        profitweekdateto: "2022-04-04 23:59:00",
      },
      orderItems: [
        {
          id: 615724,
          idProduct: 65,
          name: "Antitranspirante Rexona Motion Sense powder dry en roll on para dama 50 ml",
          Quantity: 1,
          UnitPriceInclTax: 20.9,
          PriceInclTax: 20.9,
        },
      ],
      metadata: {
        currentPage: 1,
        perPage: 1,
        totalPages: 1,
        updatedAt: "2022-05-27T18:28:48+00:00",
        nextRefresh: 3600,
      },
    };
  });

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test("ModalInvoice component should be render correctly", () => {
    const { container } = render(<ModalInvoice {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
