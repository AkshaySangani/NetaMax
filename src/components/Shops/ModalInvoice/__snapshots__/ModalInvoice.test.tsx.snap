// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalInvoice Component ModalInvoice component should be render correctly 1`] = `
<div>
  <div
    class="css-gmuwbf"
  />
</div>
`;